### PR TITLE
Add more BIC detail about PHP and PHPUnit

### PR DESCRIPTION
**Pull request by jeff-matthews**
This pull request (PR) adds more detail about the backward incompatible changes we introduced when we bumped support for PHP and PHPUnit.

Details were taken from the internal engineering.

## Affected DevDocs pages

-  https://devdocs.magento.com/guides/v2.4/release-notes/backward-incompatible-changes/index.html

whatsnew
Added [more detail](https://devdocs.magento.com/guides/v2.4/release-notes/backward-incompatible-changes/index.html) about backward incompatible changes that were introduced after adding support for PHP 7.4 and PHPUnit 9.

### CODE REVIEW

File: src/guides/v2.4/release-notes/backward-incompatible-changes/index.md [hunk 1]
```diff
@@ -130,6 +130,84 @@ These changes allow third-party developers to optimize performance by providing
...
+   PHPUnit\Framework\TestCase::setUpBeforeClass()
+   PHPUnit\Framework\TestCase::setUp()
+   PHPUnit\Framework\TestCase::assertPreConditions()
+   PHPUnit\Framework\TestCase::assertPostConditions()
+   PHPUnit\Framework\TestCase::tearDown()
+   PHPUnit\Framework\TestCase::tearDownAfterClass()
+   PHPUnit\Framework\TestCase::onNotSuccessfulTest()
+   ```
+
+-  The following method has been removed and the tests implementation should be changed
```
Comment (keharper) [inline src/guides/v2.4/release-notes/backward-incompatible-changes/index.md:174]:
```suggestion
-  The following methods have been removed, and you should change the implementation their tests:
```

File: src/guides/v2.4/release-notes/backward-incompatible-changes/index.md [hunk 2]
```diff
@@ -130,6 +130,84 @@ These changes allow third-party developers to optimize performance by providing
...
+   readAttribute()
+   getStaticAttribute()
+   getObjectAttribute()
+   ```
+
+-  The signature of `assertContains()`, `assertNotContains()`, `assertEquals()`, and `assertNotEquals()` were changed. In most cases, more specific methods should be used instead, like `assertStringContainsString()`
+
+#### Tips and Tricks
+
+-  Use `\PHPUnit\Framework\Assert::assertEqualsCanonicalizing()` if you need to compare 2 entities with a different order of elements. `assertEquals()` has been used before.
```
Comment (keharper) [inline src/guides/v2.4/release-notes/backward-incompatible-changes/index.md:208]:
```suggestion
-  Use `\PHPUnit\Framework\Assert::assertEqualsCanonicalizing()` if you need to compare two entities with a different order of elements. `assertEquals()` has been used before.
```

File: src/guides/v2.4/release-notes/backward-incompatible-changes/index.md [hunk 3]
```diff
@@ -130,6 +130,84 @@ These changes allow third-party developers to optimize performance by providing
 -  Introduced a Bulk version of `IsProductSalableForRequestedQtyInterface` API
 -  Introduced a Bulk version of `IsProductSalableInterface` when working with a list of items
 
+### PHP
+
+PHP 7.4 support is added to Magento 2.4.0 and the lowest compatible version is PHP 7.3. As the result, some of the Composer libraries used by Magento have been updated.
```
Comment (keharper) [inline src/guides/v2.4/release-notes/backward-incompatible-changes/index.md:135]:
```suggestion
PHP 7.4 support is added to Magento 2.4.0, and the lowest compatible version is PHP 7.3. As the result, some of the Composer libraries used by Magento have been updated.
```

**Comment by jeff-matthews** [comment]
running tests

**Pull request merged**